modules/client/src/main/scala/ch/epfl/bluebrain/nexus/iam/client/types/Path.scala

```diff
@@ -0,0 +1,262 @@
+package ch.epfl.bluebrain.nexus.iam.client.types
+
+import akka.http.scaladsl.model.Uri
+import cats.Show
+import io.circe.{Decoder, Encoder}
+
+import scala.annotation.tailrec
+
+/**
+  * Represents an absolute resource path as an optionally empty ordered collection of segments.  The canonical form of
+  * a path would be ''/a/b/c/d''.
+  */
+sealed abstract class Path extends Serializable {
```

Comment:
should we rename this to `Address` to avoid the collision with the RDF lib `Path`?